### PR TITLE
Made obstacle class abstract and adds extra stuff.

### PR DESCRIPTION
**Pull request by nanoterran**
This updates the obstacle class and makes it abstract. We are now able to pass in the color from the derived classes.

### CODE REVIEW

File: Source/Obstacle/Obstacle.h [hunk 1]
```diff
@@ -13,7 +13,8 @@
 enum
 {
 	maxDefence = 100,
-	minDefence = 0
+	minDefence = 0,
+	defaultDefece = 100
```
Comment (theawesomechin) [inline Source/Obstacle/Obstacle.h]: Small thing default Defense is spelled wrong. Why does an obstacle have a default defense of 100? Have we set a max or a min defense?

**Comment by theawesomechin** [comment]
This PR looks good but I also don't have much context of the whole thing. I will comeback and comment more effectively once I wrap my head around the it. Only thing is there is a bit of misspelling its not a huge deal but it helps in a professional setting when you need to search through all the files. Something I have learned over time is with easy naming conventions (which I hadn't worried about before) really speeds up production and helps you understand when you comeback later and revisit the code.

File: Source/Obstacle/Obstacle.h (hunk 1 above)
Comment (nanoterran) [reply Source/Obstacle/Obstacle.h]: I was thinking that we might have a default value in the future, so I added that to be changed in the future for a default value. I will just make it max defense.

**Comment by nanoterran** [comment]
Merging....

**Pull request merged**